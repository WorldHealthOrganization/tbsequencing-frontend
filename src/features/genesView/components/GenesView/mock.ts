export const mockGenes = [
  { geneId: 1, geneName: 'rpoC' },
  { geneId: 2, geneName: 'rpoB' },
  { geneId: 3, geneName: 'rpoA' },
  { geneId: 4, geneName: 'vpo8B' },
  { geneId: 5, geneName: 'vpo8A' },
  { geneId: 6, geneName: 'Rv0669c' },
  { geneId: 7, geneName: 'cfp21' },
  { geneId: 8, geneName: 'J8670_RS10350' },
  { geneId: 9, geneName: 'J8670_RS10355' },
];
