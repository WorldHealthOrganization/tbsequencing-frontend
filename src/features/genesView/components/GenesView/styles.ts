import { css } from '@emotion/react';
import { font } from '../../../../components/typography/fontSizes';
import appColors from '../../../../styles/colors';

export const genesViewStyles = {
  tablesWrapper: css({
    display: 'flex',
    marginTop: '16px',
    '>div': {
      marginRight: '17px',
      marginBottom: '10px',
      maxHeight: '470px',
      '&:last-of-type': {
        marginRight: 0,
      },
    },
  }),
  autoCompleteInputLabel: css({
    fontFamily: font.regular,
    fontWeight: '400',
  }),
  autoCompleteContainer: css({
    display: 'flex',
    alignItems: 'center',
  }),
  sxAutoComplete: {
    width: '300px',
    marginRight: '40px',
    '.MuiOutlinedInput-root': {
      padding: 0,
    },
  },
  overviewTitle: css({
    marginTop: '32px',
    marginBottom: '24px',
  }),
  gnomadWrapper: css({
    marginTop: '32px',
  }),
  gridWrapper: css({
    margin: '20px 0',
    paddingBottom: '40px',
  }),
};

export const geneOverviewLink = css({
  textDecoration: 'none',
  color: appColors.secondary.whoBlue,
});

export const cdsLink = css({
  width: '200px',
  overflow: 'hidden',
});

export const gridContainerStyles = css({
  maxHeight: 50 * 16, // 31 = height of single row; 16 = need to show 15 records in grid + 1(header)
  minHeight: '300px',
  height: 'unset',
  'tr th': {
    padding: '10px 16px',
  },
});

export const container = css({
  overflow: 'hidden',
});
