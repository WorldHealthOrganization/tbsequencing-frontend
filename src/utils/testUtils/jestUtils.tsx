import React, { ReactElement } from 'react';
import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { reducer } from '../../app/store';

const getWrapper = (
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
  } = {},
) => ({ children }: { children: ReactElement }) => (
  <Provider store={store}>{children}</Provider>
);

function render(
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
    ...renderOptions
  } = {},
) {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const getTreeJSONWithRedux = (ui: ReactElement, {
  // @ts-ignore
  preloadedState,
  store = configureStore({ reducer, preloadedState }),
} = {}) => {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  const tree = renderer
    .create(<Wrapper>{ui}</Wrapper>)
    .toJSON();

  return tree;
};

export const wrappedInRouter = (
  component: ReactElement,
) => <BrowserRouter>{component}</BrowserRouter>;

export * from '@testing-library/react';
export { render, getTreeJSONWithRedux };
