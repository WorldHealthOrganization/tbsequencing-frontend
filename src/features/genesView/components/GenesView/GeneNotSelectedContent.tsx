/** @jsxImportSource @emotion/react */
import React from 'react';
import { genesViewStyles } from './styles';
import AssociationTable, { LeftColumnButton } from '../AssociationTable';
import { IAssociationTableData } from '../AssociationTable/AssociationTable';

interface IGeneNotSelectedContentProps {
  isMostSearchLoading: boolean;
  isRecentSearchLoading: boolean;
  mostSearch: IAssociationTableData[]
  recentSearch: IAssociationTableData[],
  onGeneClick: (item: IAssociationTableData) => void
}

const GeneNotSelectedContent = ({
  isMostSearchLoading,
  mostSearch,
  isRecentSearchLoading,
  recentSearch,
  onGeneClick,
}: IGeneNotSelectedContentProps) => (
  <div css={genesViewStyles.tablesWrapper}>
    <AssociationTable
      isLoading={isMostSearchLoading}
      leftColumnItemComponent={LeftColumnButton}
      tableHeader="Most Searched"
      onClick={onGeneClick}
      data={mostSearch}
    />
    <AssociationTable
      isLoading={isRecentSearchLoading}
      leftColumnItemComponent={LeftColumnButton}
      tableHeader="Recently Searched"
      data={recentSearch}
      onClick={onGeneClick}
    />
  </div>
);

export default GeneNotSelectedContent;
