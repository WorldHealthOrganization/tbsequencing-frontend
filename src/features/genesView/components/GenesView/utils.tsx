/** @jsxImportSource @emotion/react */
import React from 'react';
import { formatDistance, parseISO } from 'date-fns';
import { IGenesOverview, ISearchCount, ISearchRecent } from '../../models';
import { GeneAssociationData, GeneAssociationGroupedData } from '../../../../services/drugsApi/models';
import { cdsLink, geneOverviewLink } from './styles';

const getDayByDate = (date: string) => (date.includes('hours') ? 'Today' : date);

export const formatRecentlySearch = (searchRecentlyData?: ISearchRecent[]) => {
  if (!searchRecentlyData) return [];

  return searchRecentlyData.map((item) => ({
    id: item.geneDbCrossrefId,
    leftColValue: item.geneName || item.locusTag,
    rightColValue: getDayByDate(
      `${formatDistance(parseISO(item.date as string), new Date(), { addSuffix: true })}`,
    ),
  }));
};

export const formatMostSearch = (searchMostData?: ISearchCount[]) => {
  if (!searchMostData) return [];

  return searchMostData.map((item) => ({
    id: item.geneDbCrossrefId,
    leftColValue: item.geneName || item.locusTag,
    rightColValue: item.counter,
  }));
};

export const formatDrugsTableGenesView = (data?: any[]) => {
  if (!data) return [];

  return data.map((item) => ({
    id: item.drugId,
    leftColValue: item.code,
    rightColValue: item.drugName,
  }));
};

export const formatResistantDataGrouped = (data?: GeneAssociationGroupedData[]) => {
  if (!data?.length) return [];

  return data.map((item) => ({
    id: item.geneDbCrossrefId,
    leftColValue: item.geneName || item.locusTag,
    tier: item.tier,
  }));
};

export const formatResistantData = (data?: GeneAssociationData[]) => {
  if (!data?.length) return [];

  return data.map((item) => ({
    id: item.geneDbCrossref,
    leftColValue: item.geneName,
    rightColValue: item.drugs.map((d) => d.code).join(', '),
  }));
};

export const getGenomeContextLength = (gene?: IGenesOverview) => {
  if (!gene) return {};
  const length = gene.endPos - gene.startPos;
  const startPos = parseInt(String(gene.startPos - length * 1.5), 10);
  const endPos = parseInt(String(gene.endPos + length * 1.5), 10);
  return {
    startPos: startPos > 0 ? startPos : 1,
    endPos,
  };
};

export const formatGenesOverviewData = (data?: IGenesOverview) => {
  if (!data) return [];
  const {
    geneName,
    locusTag,
    geneType,
    geneDescription,
    startPos,
    endPos,
    ncbiId,
    proteinLength,
  } = data;

  return [
    { title: 'Gene Symbol', value: geneName || locusTag },
    {
      title: 'NCBI Gene ID',
      value: ncbiId,
      element: (
        <a
          css={geneOverviewLink}
          target="_blank"
          href={`https://www.ncbi.nlm.nih.gov/gene/${ncbiId}`}
          rel="noreferrer"
        >
          {ncbiId}
        </a>
      ),
    },
    { title: 'Gene Description', value: geneDescription },
    { title: 'Locus Tag', value: locusTag },
    { title: 'Gene type', value: geneType ? 'protein coding' : null },
    {
      title: 'Location',
      value: endPos && startPos
        ? `${startPos.toLocaleString('en-US')}..${endPos.toLocaleString('en-US')}`
        : null,
    },
    {
      title: 'Length',
      value: endPos && startPos
        ? `${(endPos - startPos).toLocaleString('en-US')} nt`
        : null,
    },
    {
      title: 'CDS',
      value: geneType,
      element: (
        <a
          css={[geneOverviewLink, cdsLink]}
          target="_blank"
          href={`https://www.ncbi.nlm.nih.gov/protein/${geneType}`}
          rel="noreferrer"
        >
          {geneType}
        </a>
      ),
    },
    {
      title: 'Protein Length',
      value: proteinLength
        ? `${proteinLength.toLocaleString('en-US')} aa`
        : null,
    },
  ];
};
