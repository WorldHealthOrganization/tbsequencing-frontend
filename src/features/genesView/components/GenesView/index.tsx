import GenesView from './GenesView';

export default GenesView;
