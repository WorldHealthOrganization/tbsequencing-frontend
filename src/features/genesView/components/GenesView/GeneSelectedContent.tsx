/** @jsxImportSource @emotion/react */
import React, { useEffect, useState } from 'react';
import GenomeContextWrapper from '../GenomeContextWrapper';
import H2 from '../../../../components/typography/H2';
import AssociationTable, { LeftColumnButton } from '../AssociationTable';
import { container, genesViewStyles, gridContainerStyles } from './styles';
import GnomadVariants from '../../../../components/GnomadVariants';
import { IGene, IVariant } from '../../../../services/genesApi/models';
import {
  useGetGenesByIdQuery,
  useGetGenesTableDataQuery,
  useGetGenomeContextQuery,
} from '../../../../services/genesApi/genesApi';
import { useTableData } from '../../../drugsView/hooks/useTableData';
import DataGrid from '../../../../components/DataGrid';
import {
  GeneTableColumns,
  GridType,
  IColumn,
  ScrollType,
} from '../../../../components/DataGrid/models';
import { IVariantLocal } from '../../models';
import { formatDrugsTableGenesView, formatGenesOverviewData, getGenomeContextLength } from './utils';
import GeneOverview from '../GeneOverview';
import { IDrug, TableData } from '../../../../services/drugsApi/models';
import { IAssociationTableData } from '../AssociationTable/AssociationTable';
import { useNavigateWithParams } from '../../../../hooks/useNavigateWithParams';
import { useGetGeneAssociationsByIDQuery } from '../../../../services/drugsApi/drugsApi';

interface IGeneSelectedContentProps {
  selectedGene: IGene
}

const columns: IColumn<GeneTableColumns>[] = [
  { name: 'variantName', header: 'Variant' },
  { name: 'nucleodicAnnName', header: 'Nucleic Annotation' },
  { name: 'proteicAnnName', header: 'Proteic Annotation' },
  { name: 'consequence', header: 'Consequence' },
  { name: 'variantGrade', header: 'Grade' },
  { name: 'globalFrequency', header: 'Global Frequency' },
  { name: 'totalCounts', header: 'Total Samples' },
  { name: 'resistantCount', header: 'Resistant Count' },
  { name: 'susceptbleCount', header: 'Susceptible Count' },
  { name: 'intermediateCount', header: 'Intermediate Count' },
];

const columnsWithoutStats: IColumn<GeneTableColumns>[] = [
  { name: 'variantName', header: 'Variant' },
  { name: 'nucleodicAnnName', header: 'Nucleic Annotation' },
  { name: 'proteicAnnName', header: 'Proteic Annotation' },
  { name: 'consequence', header: 'Consequence' },
  { name: 'variantGrade', header: 'Grade' },
  { name: 'globalFrequency', header: 'Global Frequency' },
  { name: 'totalCounts', header: 'Total Samples' },
];

const GeneSelectedContent = ({ selectedGene }: IGeneSelectedContentProps) => {
  const navigate = useNavigateWithParams();
  const onClick = (item: IAssociationTableData) => {
    const params = new URLSearchParams({
      drugId: String(item.id),
      code: String(item.leftColValue),
      drugName: String(item.rightColValue),
    });
    navigate('/drugs', params);
  };

  const [selectedVariant, setSelectedVariant] = useState<IVariantLocal | {}>({});
  const [activeIndex, setActiveIndex] = useState<number>(-1);
  const [gridViewPortData, setGridViewPortData] = useState<number[]>([]);
  const [selectedResistantDrug, setSelectedResistantDrug] = useState<IDrug>();

  const { data: genesDetails, isLoading: isGeneDetailsLoading } = useGetGenesByIdQuery({
    geneDbCrossrefId: selectedGene.geneDbCrossrefId,
  });

  const { startPos, endPos } = getGenomeContextLength(genesDetails?.genesOverview);

  const { data: genomeContextData, isLoading: isGenomeContextLoading } = useGetGenomeContextQuery({
    startPos,
    endPos,
  }, { skip: !genesDetails?.genesOverview?.startPos || !genesDetails?.genesOverview?.endPos });

  const {
    data: associationsData,
    isLoading: isAssociationsLoading,
    isSuccess,
  } = useGetGeneAssociationsByIDQuery({
    geneDbCrossrefId: selectedGene.geneDbCrossrefId,
  });

  const {
    tableData = [],
    isTableDataLoading,
    filters,
    updateFilters,
  } = useTableData({
    useGetDataQuery: useGetGenesTableDataQuery,
    geneDbCrossrefId: selectedGene.geneDbCrossrefId,
    drugID: selectedResistantDrug?.drugId || associationsData?.[0]?.drugId,
    skipQuery: !isSuccess,
  });

  const drugsTableData = formatDrugsTableGenesView(associationsData);
  const genesOverviewFormattedData = formatGenesOverviewData(genesDetails?.genesOverview);

  useEffect(() => {
    const newActiveIndex = tableData.findIndex((data: TableData) => (
      data.variantName === (selectedVariant as IVariantLocal).variantName
    ));
    setActiveIndex(newActiveIndex);
  }, [tableData, selectedVariant]);

  const onVariantClick = (variant: any) => {
    setSelectedVariant(variant);
  };

  const onScroll = (visibleRowsRange: number[]) => {
    setGridViewPortData(visibleRowsRange);
  };

  const onChangeResistantDrug = (value: number) => {
    if (!associationsData?.length) return;
    const drug = associationsData.find((item) => item.drugId === value);
    setSelectedResistantDrug(drug);
  };

  return (
    <div css={container}>
      <GenomeContextWrapper
        selectedGene={genesDetails?.genesOverview}
        isLoading={isGenomeContextLoading}
        data={genomeContextData}
        maxDomain={endPos}
        minDomain={startPos}
      />
      <H2 style={genesViewStyles.overviewTitle}>Overview</H2>
      <div css={genesViewStyles.tablesWrapper}>
        <GeneOverview data={genesOverviewFormattedData} isLoading={isGeneDetailsLoading} />
        <AssociationTable
          tableHeader="Gene-drug resistance associations"
          data={drugsTableData}
          isLoading={isAssociationsLoading}
          leftColumnItemComponent={LeftColumnButton}
          onClick={onClick}
          noDataText="No associated data with this gene"
        />
      </div>
      <div css={genesViewStyles.gnomadWrapper}>
        <GnomadVariants
          selectedGene={selectedGene}
          variants={tableData as unknown as IVariant[]}
          isLoading={isTableDataLoading}
          variantsInTable={gridViewPortData}
          onClick={onVariantClick}
        />
      </div>
      <div css={genesViewStyles.gridWrapper}>
        <DataGrid<GeneTableColumns>
          containerStyles={gridContainerStyles}
          columns={associationsData?.length ? columns : columnsWithoutStats}
          isLoading={isTableDataLoading}
          filterValues={filters}
          onFilterChange={updateFilters}
          dataSource={tableData}
          selectedDropdownItem={selectedGene}
          selectedRow={activeIndex}
          type={GridType.Gene}
          scrollType={ScrollType.Infinite}
          resistantDrugs={associationsData}
          selectedResistantDrug={selectedResistantDrug || associationsData?.[0]}
          onChangeResistantDrug={onChangeResistantDrug}
          onScroll={onScroll}
        />
      </div>
    </div>
  );
};

export default GeneSelectedContent;
