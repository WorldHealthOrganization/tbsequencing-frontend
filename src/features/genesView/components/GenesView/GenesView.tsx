/** @jsxImportSource @emotion/react */
import React, { SyntheticEvent, useEffect, useState } from 'react';
import { useSearchParams } from 'react-router-dom';
import BasePage from '../../../../components/BasePage';
import AppPaper from '../../../../components/AppPaper';
import AutocompleteInput from '../../../../components/AutocompleteInput';
import { genesViewStyles } from './styles';
import H3 from '../../../../components/typography/H3';
import {
  useGetMostSearchHistoryQuery,
  useGetRecentSearchHistoryQuery,
} from '../../../../services/genesApi/genesApi';
import { useAutocompleteQuery } from '../../hooks/useAutocompleteQuery';
import { useDebounce } from '../../../../hooks/useDebounce';
import GeneNotSelectedContent from './GeneNotSelectedContent';
import GeneSelectedContent from './GeneSelectedContent';
import { IGene } from '../../../../services/genesApi/models';
import { formatMostSearch, formatRecentlySearch } from './utils';

const GenesView = () => {
  const [searchParams] = useSearchParams();

  const defaultGene: IGene = {
    geneDbCrossrefId: Number(searchParams.get('geneDbCrossrefId')),
    geneName: searchParams.get('geneName'),
    locusTag: null,
    id: Number(searchParams.get('geneDbCrossrefId')),
    endPos: 1,
    startPos: 0,
    strand: -1,
  };

  const [inputValue, setInputValue] = useState('');
  const [
    selectedValue,
    setSelectedValue,
  ] = useState<IGene | null>(defaultGene.geneDbCrossrefId ? defaultGene : null);
  const debouncedValue = useDebounce<string>(inputValue, 500);

  const {
    data: mostSearch = [],
    isLoading: isMostSearchLoading, refetch: mostSearchHistoryRefetch,
  } = useGetMostSearchHistoryQuery(undefined, { refetchOnMountOrArgChange: true });

  const {
    data: recentSearch = [],
    isLoading: isRecentSearchLoading, refetch: resentSearchRefetch,
  } = useGetRecentSearchHistoryQuery(undefined, { refetchOnMountOrArgChange: true });

  const {
    data: genes = [], fetchData, isLoading,
  } = useAutocompleteQuery();

  const getOptions = () => {
    const inputEmpty = !inputValue;
    const mostSearchedData = mostSearch
      .slice(-10)
      .sort((prev, next) => next.counter - prev.counter);

    return inputEmpty ? mostSearchedData : genes;
  };

  const options = getOptions();

  const recentSearchLocal = formatRecentlySearch(recentSearch);
  const mostSearchLocal = formatMostSearch(mostSearch);

  const handleInputChange = (e: SyntheticEvent, value: string) => {
    setInputValue(value);
  };

  useEffect(() => {
    if (debouncedValue !== '') {
      fetchData({ search: debouncedValue });
    }
  }, [debouncedValue]);

  const patchedSelectedGene = genes.length === 1
    ? { ...selectedValue, ...genes[0] }
    : selectedValue;

  return (
    <BasePage pageHeader="Gene View">
      <AppPaper>
        <div css={genesViewStyles.autoCompleteContainer}>
          <AutocompleteInput
            isLoading={isLoading}
            onSelectionChanged={(e, value) => {
              if (value === null) {
                mostSearchHistoryRefetch();
                resentSearchRefetch();
              }

              setSelectedValue(value);
            }}
            selectedValue={selectedValue}
            onChange={handleInputChange}
            value={inputValue}
            options={options}
            style={genesViewStyles.sxAutoComplete}
          />
          <H3
            style={genesViewStyles.autoCompleteInputLabel}
          >
            Search Genes by Gene Symbol, NCBI ID or Locus Tag
          </H3>
        </div>
      </AppPaper>
      {
        !selectedValue
          ? (
            <GeneNotSelectedContent
              isMostSearchLoading={isMostSearchLoading}
              isRecentSearchLoading={isRecentSearchLoading}
              mostSearch={mostSearchLocal}
              recentSearch={recentSearchLocal}
              onGeneClick={(item) => {
                setSelectedValue({
                  endPos: 0,
                  startPos: 0,
                  strand: -1,
                  id: item.id,
                  geneDbCrossrefId: item.id,
                  geneName: item.leftColValue as string,
                  locusTag: item.leftColValue as string,
                });
              }}
            />
          )
          : <GeneSelectedContent selectedGene={(patchedSelectedGene) as IGene} />
      }
    </BasePage>
  );
};

export default GenesView;
