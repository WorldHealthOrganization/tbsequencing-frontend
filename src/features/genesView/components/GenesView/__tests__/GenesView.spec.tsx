import React from 'react';
import { getTreeJSONWithRedux, render, wrappedInRouter } from '../../../../../utils/testUtils/jestUtils';
import GenesView from '../index';

describe('GenesView', () => {
  test('renders GenesView component without error', () => {
    render(
      wrappedInRouter(<GenesView />),
    );
  });

  test('Should render correctly', () => {
    const tree = getTreeJSONWithRedux(
      wrappedInRouter(<GenesView />),
    );

    expect(tree).toMatchSnapshot();
  });
});
