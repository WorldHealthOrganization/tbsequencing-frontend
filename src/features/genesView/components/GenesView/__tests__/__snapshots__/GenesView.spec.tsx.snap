// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenesView Should render correctly 1`] = `
<div
  className="css-xn17ji-BasePage"
>
  <p
    className="MuiTypography-root MuiTypography-body1 css-177n4ga-MuiTypography-root-BaseText"
  >
    Gene View
  </p>
  <div
    className="css-1y50k9a-AppPaper"
  >
    <div
      className="css-1a6j5wi-GenesView"
    >
      <div
        aria-owns={null}
        className="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-16awh2u-MuiAutocomplete-root"
        data-testid="autocomplete"
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
      >
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-z2zmqk-MuiFormControl-root-MuiTextField-root"
        >
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-activedescendant={null}
              aria-autocomplete="both"
              aria-expanded={false}
              aria-invalid={false}
              autoCapitalize="none"
              autoComplete="off"
              autoFocus={false}
              className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id=":r0:"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              placeholder="Search for gene..."
              readOnly={false}
              required={false}
              role="combobox"
              spellCheck="false"
              type="text"
              value=""
            />
            <div
              className="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
            >
              <button
                aria-label="Open"
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
                title="Open"
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </button>
            </div>
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-hdw1oc"
              >
                <span
                  className="notranslate"
                >
                  â€‹
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <p
        className="MuiTypography-root MuiTypography-body1 css-m8wdzz-MuiTypography-root-BaseText"
      >
        Search Genes by Gene Symbol, NCBI ID or Locus Tag
      </p>
    </div>
  </div>
  <div
    className="css-1k0mzhl-GeneNotSelectedContent"
  >
    <div
      className="css-9j8ib4-AppPaper"
    >
      <div>
        <p
          className="MuiTypography-root MuiTypography-body1 css-5th3yp-MuiTypography-root-BaseText"
        >
          Most Searched
        </p>
      </div>
      <span
        className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1ag5kwf-MuiCircularProgress-root"
        role="progressbar"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx={44}
            cy={44}
            fill="none"
            r={20.2}
            strokeWidth={3.6}
            style={Object {}}
          />
        </svg>
      </span>
    </div>
    <div
      className="css-9j8ib4-AppPaper"
    >
      <div>
        <p
          className="MuiTypography-root MuiTypography-body1 css-5th3yp-MuiTypography-root-BaseText"
        >
          Recently Searched
        </p>
      </div>
      <span
        className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1ag5kwf-MuiCircularProgress-root"
        role="progressbar"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx={44}
            cy={44}
            fill="none"
            r={20.2}
            strokeWidth={3.6}
            style={Object {}}
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;
